.github/workflows/changelog-check.yml

```
@@ -0,0 +1,88 @@
+name: Changelog Check
+
+on:
+  workflow_call:
+    inputs:
+      base-branch:
+        required: false
+        type: string
+      head-ref:
+        required: true
+        type: string
+      pr-number:
+        description: 'The pull request number'
+        required: true
+        type: string
+      repo:
+        description: 'The repository to check'
+        required: true
+        type: string
+    secrets:
+      gh-token:
+        required: true
+
+jobs:
+  check-changelog:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out target repository
+        uses: actions/checkout@v4
+        with:
+          repository: ${{ inputs.repo }}
+          ref: ${{ inputs.head-ref }}
+          path: target-repo
+          fetch-depth: 0
+
+      - name: Checkout github-tools repository
+        uses: actions/checkout@v4
+        with:
+          repository: MetaMask/github-tools
+          ref: cryptodev2s/changelog-checker
+          path: github-tools
+
+      - name: Enable Corepack
+        run: corepack enable
+        shell: bash
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: ./github-tools/.nvmrc
+          cache-dependency-path: ./github-tools/yarn.lock
+          cache: yarn
+
+      - name: Install dependencies
+        run: yarn --immutable
+        shell: bash
+        working-directory: ./github-tools
+
+      - name: Check PR Labels
+        id: label-check
+        env:
+          GITHUB_TOKEN: ${{ secrets.gh-token }}
+          GITHUB_REPO: ${{ inputs.repo }}
+          PR_NUMBER: ${{ inputs.pr-number }}
+        run: |
+          # Fetch labels and check for no-changelog
+          if labels=$(curl -s -H "Authorization: Bearer $GITHUB_TOKEN" \
+            "https://api.github.com/repos/${GITHUB_REPO}/issues/${PR_NUMBER}/labels"); then
+            if echo "$labels" | jq -e '.[] | select(.name == "no-changelog")' > /dev/null; then
+              echo "no-changelog label found, skipping changelog check."
+              echo "skip_check=true" >> "$GITHUB_OUTPUT"
+            else
+              echo "No no-changelog label found, proceeding with check."
+              echo "skip_check=false" >> "$GITHUB_OUTPUT"
+            fi
+          else
+            echo "::error::Failed to fetch labels. Assuming changelog check is needed."
+            echo "skip_check=false" >> "$GITHUB_OUTPUT"
+          fi
+        shell: bash
```

Nit: we can also do the following and avoid the api call
```suggestion
      - name: Check PR Labels
        id: label-check
        run: |
          if echo '${{ inputs.labels }}' | jq -e '.[] | select(.name == "no-changelog")' > /dev/null; then
            echo "no-changelog label found, skipping changelog check."
            echo "skip_check=true" >> "$GITHUB_OUTPUT"
          else
            echo "No no-changelog label found, proceeding with check."
            echo "skip_check=false" >> "$GITHUB_OUTPUT"
          fi
        shell: bash